source/server-discovery-and-monitoring/server-discovery-and-monitoring.md

```diff
@@ -863,38 +870,46 @@ else if topologyDescription.setName != serverDescription.setName:
 
 if serverDescription.maxWireVersion >= 17:  # MongoDB 6.0+
     # Null values for both electionId and setVersion are always considered less than
-    if serverDescription.electionId > topologyDescription.maxElectionId or (
-        serverDescription.electionId == topologyDescription.maxElectionId
-        and serverDescription.setVersion >= topologyDescription.maxSetVersion
-    ):
-        topologyDescription.maxElectionId = serverDescription.electionId
-        topologyDescription.maxSetVersion = serverDescription.setVersion
-    else:
-        # Stale primary.
-        # replace serverDescription with a default ServerDescription of type "Unknown"
+    if serverDescription.electionId < topologyDescription.maxElectionId:
```

Comment:
I see we the pseudocode is a bit more verbose now that the logic needs to distinguish between the electionId vs setVersion stale cases. What do you think about one error message that include the two tuples:
```
"primary marked stale due to electionId/setVersion mismatch, <stale tuple> is stale compared to <max tuple>"
```

That way we don't need to refactor this code and more importantly drivers don't need to refactor their comparison logic.